src/core/storage/sframe_interface/unity_sframe.cpp

```diff
@@ -1542,23 +1585,49 @@ std::list<std::shared_ptr<unity_sframe_base>> unity_sframe::drop_missing_values(
 
   std::function<flexible_type(const sframe_rows::row&)> filter_fn;
   if (all) {
-    filter_fn = [column_indices](const sframe_rows::row& row)->flexible_type {
-                                  size_t num_missing_values = 0;
-                                  for(const auto &i : column_indices) {
-                                    if(row[i].is_na()) {
-                                      ++num_missing_values;
-                                    }
-                                  }
-                                  return (num_missing_values != column_indices.size());
-                                };
+    // for perf, I choose not to use std::function to wrap _contains_nan or is_nan
+    if (recursive) {
```

Comment:
After talking with @ylow, I hesitate to use std::function to wrap is_na and _contains_nan. I dup the code to avoid extra virtual calls on each cell.

Reply:
I wouldn't worry about it unless a perf profile suggests otherwise. Premature optimization is not a good thing.

Reply:
I think it's ok to dup the code for this simple case (if we'd like to provide more options in the future, we can do refactor later). The optimization is to conduct if-else check once to register the right func for the filter before the filter is invoked for thousands of rows from a sframe, without paying the overhead of virtual function call for each row.